apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts

```
@@ -1918,55 +1871,27 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
-                            return item;
-                        });
-                    });
-                } else {
-                    (Array.isArray(this.currentTxn.selectedAccount.accountApplicableDiscounts) ? this.currentTxn.selectedAccount.accountApplicableDiscounts : []).forEach(element => {
-                        this.currentTxn?.discounts?.map(item => {
-                            if (element?.uniqueName === item?.discountUniqueName) {
-                                item.isActive = true;
-                            }
-                            return item;
-                        });
+                (Array.isArray(this.currentTxn?.selectedAccount.accountApplicableDiscounts) ? this.currentTxn?.selectedAccount.accountApplicableDiscounts : []).forEach(element => {
```

Biome (2.1.2)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)